src/storage-client/src/sink.rs

```diff
@@ -179,9 +170,18 @@ pub struct TopicConfig {
 #[derive(Debug, Clone)]
 pub enum TopicCleanupPolicy {
     /// Clean up the topic using a time and/or size based retention policies.
-    ///
-    /// Use `-1` to indicate infinite retention.
-    Retention(KafkaSinkConnectionRetention),
+    Retention {
+        /// A time-based retention policy.
+        ///
+        /// `None` indicates broker default. `Some(-1)` indicates infinite
+        /// retention.
+        ms: Option<i64>,
```

Comment:
This would also let us use `Value(Duration)` to make these options more consistent with other interval/duration-based settings.

Reply:
Ah, so if it's okay I'd love to punt this to a future PR (but probably not one I'll have time to submit myself). Resolving this in a way that's consistent across the part count, replication factor, and retention options is tricky because e.g. the max part count is `i32::MAX`, which is less than `u32::MAX`. (We just got a `NonNeg<i32>` that could work nicely here...)

I feel like https://github.com/MaterializeInc/materialize/issues/23794 will be a good time to clean this up, as we'll be figuring out how to represent the difference between "unset", "broker default", "infinite", and "finite with value &lt;v>" in SQL, and then can carry through that improvement through the Rust code. For now the function is private storage API so I feel like it's not too bad to have it be a lower level function that more closely mirrors the underlying librdkafka APIs.

Reply:
Sounds good